pkg/openstack/common.go

```diff
@@ -7,12 +7,32 @@ import (
 
 	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
 	"github.com/openstack-k8s-operators/lib-common/modules/common/util"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
 	apimeta "k8s.io/apimachinery/pkg/api/meta"
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// CheckDeleteDisabled - Delete the object which in turn will clean the sub resources
+func CheckDeleteDisabled(ctx context.Context, helper *helper.Helper, obj client.Object) (ctrl.Result, error) {
```

Comment:
I would not call it `Check` as it actively deletes not just checks for a condition. It also does not check for Disabled. So I would call this simply `Delete()`

Reply:
`ensureDeleted` would work for me to encode that this function works if the service is already deleted. Still the function does not depend on the service being disabled so I would not add that to the name. But we can agree to disagree here.